src/test/java/com/google/step/servlets/PromoCodeServletTest.java

```diff
@@ -1,12 +1,128 @@
+import com.google.api.services.youtube.YouTube.Channels;
+import com.google.api.services.youtube.YouTube.PlaylistItems;
+import com.google.api.services.youtube.YouTube;
+
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
 
-// @RunWith(JUnit4.class)
+@RunWith(MockitoJUnitRunner.class)
 public final class PromoCodeServletTest {
-    // TODO(margaret): Implementation coming
-}
+        
+    private PromoCodeServlet servlet;
+
+    private static final String CHANNEL_ID_NONEXISTENT = "CHANNEL_ID_NONEXISTENT";
+    private static final String UPLOAD_ID_NONEXISTENT = "UPLOAD_ID_NONEXISTENT";
+    private static final String CHANNEL_ID = "CHANNEL_ID";
+    private static final String UPLOAD_ID = "UPLOAD_ID";
+    private static final String VIDEO_ID = "VIDEO_ID";
+    private static final String IOEXCEPTION = "IOEXCEPTION";
+    private static final long MOCK_DATE = 0L;
+
+    @Mock
+    HttpServletRequest request;
+ 
+    @Mock
+    HttpServletResponse response;
+
+    @Mock
+    YoutubeInfoScraper infoScraper;
+
+    @Before
+    public void setup() throws IOException {
+        //MockitoAnnotations.initMocks(this);
+        
+        infoScraper = mock(YoutubeInfoScraper.class);
+        servlet = new PromoCodeServlet(infoScraper);
+    }
+
+    @Test
+    public void incorrectChannelIdRequest() throws ServletException, IOException {
+        when(request.getParameter("formInput")).thenReturn(CHANNEL_ID_NONEXISTENT);
+        
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
 
+        when(response.getWriter()).thenReturn(pw);
+        when(infoScraper.scrapeChannelUploadPlaylist(anyString())).thenReturn(Optional.empty());
+
+        servlet.doGet(request, response);
+        String result = sw.getBuffer().toString();
+
+        assertThat(result , equalTo("false\n"));
+    }
+
+    @Test
+    public void correctChannelIdRequest() throws ServletException, IOException {
+        when(request.getParameter("formInput")).thenReturn(CHANNEL_ID);
+        
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+
+        when(response.getWriter()).thenReturn(pw);
+        when(infoScraper.scrapeChannelUploadPlaylist(CHANNEL_ID)).thenReturn(Optional.of(UPLOAD_ID));
+        when(infoScraper.scrapePromoCodesFromPlaylist(UPLOAD_ID)).thenReturn(Optional.of(Arrays.asList(
+                PromoCode.create("LINUS", VIDEO_ID, new Date(MOCK_DATE)))));
+        servlet.doGet(request, response);
+        String result = sw.getBuffer().toString();
+
+        assertThat(result , equalTo("[{\"promoCode\":\"LINUS\",\"videoId\":\"VIDEO_ID\",\"videoUploadDate\":\"Jan 1, 1970, 12:00:00 AM\"}]\n"));
```

Comment:
Is there a better way to do this? This seems kind of hacky.

Reply:
Because I'm comparing strings I don't think there's a better way to do it.